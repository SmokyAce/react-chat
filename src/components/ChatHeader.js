import React from 'react';
import { withStyles } from 'material-ui/styles';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';
import Avatar from './Avatar';
import ChatMenu from './ChatMenu';
import UserMenu from './UserMenu';

const styles = theme => ({
  appBar: {
    position: 'fixed',
    width: `calc(100% - 320px)`
  },
  title: {
    flex: 1,
    marginLeft: theme.spacing.unit * 2
  }
});

const ChatHeader = ({ classes, actions, user, activeChat, isConnected }) => {
  return (
    <AppBar color='primary' className={classes.appBar}>
      <Toolbar>
        {activeChat ? (
          <React.Fragment>
            <Avatar colorFrom={activeChat.title}>{activeChat.title}</Avatar>
            <Typography
              variant='title'
              color='inherit'
              className={classes.title}
            >
              {activeChat.title}
              {(user.isCreator || user.isMember) && (
                <ChatMenu
                  user={user}
                  actions={actions}
                  isConnected={isConnected}
                />
              )}
            </Typography>
          </React.Fragment>
        ) : (
          <Typography variant='title' color='inherit' className={classes.title}>
            DogeCodes React Chat
          </Typography>
        )}
        <UserMenu actions={actions} user={user} isConnected={isConnected} />
      </Toolbar>
    </AppBar>
  );
};

export default withStyles(styles)(ChatHeader);
