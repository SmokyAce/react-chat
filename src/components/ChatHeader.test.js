/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatHeader from './ChatHeader';

jest.mock('./Avatar', () => () => 'Avatar');
jest.mock('./ChatMenu', () => () => 'ChatMenu');
jest.mock('./UserMenu', () => () => 'UserMenu');

const mockProps = {
  user: {
    username: 'test',
    isMember: true,
    isCreator: true,
  },
  activeChat: {
    _id: '1',
    title: 'test',
  },
  messages: [
    {
      chatId: '123',
      content: 'content',
      sender: {},
      createdAt: '2018-03-08T13:57:58.706Z',
    },
  ],
  isConnected: true,
  actions: {
    logout: jest.fn(),
    leaveChat: jest.fn(),
    deleteChat: jest.fn(),
    editUser: jest.fn(),
  },
  hasShift: false,
};

describe('<ChatHeader />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatHeader {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<ChatHeader {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without active chat', () => {
    const tree = renderer.create(<ChatHeader {...mockProps} activeChat={null} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
