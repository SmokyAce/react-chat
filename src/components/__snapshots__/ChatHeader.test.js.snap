// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatHeader /> renders correctly 1`] = `
<header
  className="MuiPaper-root-102 MuiPaper-shadow4-108 MuiAppBar-root-94 MuiAppBar-positionFixed-95 MuiAppBar-colorPrimary-100 mui-fixed ChatHeader-appBar-89"
>
  <div
    className="MuiToolbar-root-129 MuiToolbar-gutters-130 ChatHeader-toolbar-91"
  >
    <button
      aria-label="open drawer"
      className="MuiButtonBase-root-137 MuiIconButton-root-131 MuiIconButton-colorInherit-132"
      disabled={false}
      onBlur={[Function]}
      onClick={undefined}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label-136"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-139 MuiSvgIcon-fontSize-145"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-146"
      />
    </button>
    Avatar
    <h2
      className="MuiTypography-root-153 MuiTypography-title-159 MuiTypography-colorInherit-172 ChatHeader-title-92"
    >
      test
      ChatMenu
    </h2>
    UserMenu
  </div>
</header>
`;

exports[`<ChatHeader /> renders correctly without active chat 1`] = `
<header
  className="MuiPaper-root-102 MuiPaper-shadow4-108 MuiAppBar-root-94 MuiAppBar-positionFixed-95 MuiAppBar-colorPrimary-100 mui-fixed ChatHeader-appBar-89"
>
  <div
    className="MuiToolbar-root-129 MuiToolbar-gutters-130 ChatHeader-toolbar-91"
  >
    <button
      aria-label="open drawer"
      className="MuiButtonBase-root-137 MuiIconButton-root-131 MuiIconButton-colorInherit-132"
      disabled={false}
      onBlur={[Function]}
      onClick={undefined}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label-136"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-139 MuiSvgIcon-fontSize-145"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-146"
      />
    </button>
    <h2
      className="MuiTypography-root-153 MuiTypography-title-159 MuiTypography-colorInherit-172 ChatHeader-title-92"
    >
      DogeCodes React Chat
    </h2>
    UserMenu
  </div>
</header>
`;
